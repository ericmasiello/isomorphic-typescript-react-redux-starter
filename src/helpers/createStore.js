import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import axios from 'axios';
import reducers from '../client/reducers';

export default req => {
  const axiosInstance = axios.create({
    baseURL: 'http://react-ssr-api.herokuapp.com',
    headers: { cookie: req.get('cookie') || '' }
  });

  const store = createStore(
    reducers,
    {},
    applyMiddleware(thunk.withExtraArgument(axiosInstance))
  );

  return store;
};
