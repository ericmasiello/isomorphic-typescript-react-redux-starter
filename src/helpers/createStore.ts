import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import axios from 'axios';
import { Request } from 'express';
import reducers from '../client/reducers';

export default (req: Request) => {
  const axiosInstance = axios.create({
    baseURL: 'http://react-ssr-api.herokuapp.com',
    headers: { cookie: req.get('cookie') || '' },
  });

  const store = createStore(
    reducers,
    {
      users: [],
      admins: [],
      auth: null,
    },
    applyMiddleware(thunk.withExtraArgument(axiosInstance)),
  );

  return store;
};
