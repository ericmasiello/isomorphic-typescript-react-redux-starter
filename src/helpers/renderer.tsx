import * as React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import * as serialize from 'serialize-javascript';
import { Helmet } from 'react-helmet';
import { Request } from 'express';
import { Store } from 'redux';
import { ServerStyleSheet } from 'styled-components';
import Routes from '../client/Routes';

export default (req: Request, store: Store<AppState>, context: object) => {

  const app = (
    <Provider store={store}>
      <StaticRouter location={req.path} context={context}>
        <div>{renderRoutes(Routes)}</div>
      </StaticRouter>
    </Provider>
  );

  const sheet = new ServerStyleSheet();
  const content = renderToString(sheet.collectStyles(app));
  const styleTags = sheet.getStyleTags();
  const helmet = Helmet.renderStatic();

  /* tslint:disable max-line-length */
  return `
    <html>
      <head>
        ${helmet.title.toString()}
        ${helmet.meta.toString()}
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css" />
        ${styleTags}
      </head>
      <body>
        <div id="root">${content}</div>
        <script>
          window.INITIAL_STATE = ${serialize(store.getState())}
        </script>
        <script src="vendor.js"></script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;
  /* tslint:enable max-line-length */
};
