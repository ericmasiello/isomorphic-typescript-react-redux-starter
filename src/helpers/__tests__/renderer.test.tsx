import renderer from '../renderer';
import * as express from 'express';
import createStore from '../createStore';
jest.mock('react-helmet', () => {
  const obj: any = { Helmet: {} };
  obj.Helmet.renderStatic = jest.fn(() => ({
    title: { toString: jest.fn(() => 'The title') },
    meta: { toString: jest.fn(() => 'The meta') },
  }));
  return obj;
});

test('should render an html string', () => {
  const req = {} as express.Request;
  req.get = jest.fn();
  const store = createStore(req);
  const result = renderer(req, store, {});
  expect(result).toMatchSnapshot();
});
