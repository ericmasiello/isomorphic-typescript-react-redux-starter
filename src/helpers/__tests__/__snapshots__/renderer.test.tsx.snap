// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an html string 1`] = `
"
    <html>
      <head>
        The title
        The meta
        <link rel=\\"stylesheet\\" href=\\"https://cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css\\" />
        
      </head>
      <body>
        <div id=\\"root\\"><div data-reactroot=\\"\\"></div></div>
        <script>
          window.INITIAL_STATE = {\\"users\\":[],\\"admins\\":[],\\"auth\\":null}
        </script>
        <script src=\\"bundle.js\\"></script>
      </body>
    </html>
  "
`;
